### PR TITLE
api(ticdc): forward changefeed related api to changefeed owner capture

### LINKED ISSUE #9411: api(ticdc): forward changefeed related api to changefeed owner capture
api(ticdc): forward changefeed related api to changefeed owner capture

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #9411

### What is changed and how it works?
forward changefeed related api to changefeed owner capture

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/api/middleware/middleware.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/pingcap/tiflow/cdc/capture"
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/pkg/errors"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
```
Comment (CharlesCheung96) [inline cdc/api/middleware/middleware.go:26]: duplicate import
```suggestion
	cerror "github.com/pingcap/tiflow/pkg/errors"
```

File: cdc/api/middleware/middleware.go [hunk 2]
```diff
@@ -99,21 +100,13 @@ func ForwardToControllerMiddleware(p capture.Capture) gin.HandlerFunc {
...
-				ctx.Next()
-				return
-			}
+		changefeedID := changefeedIDFunc(ctx)
+		// check if this capture is the changefeed owner
+		if handleRequestIfIsChnagefeedOwner(ctx, p, changefeedID) {
+			return
 		}
 
 		// forward to the controller to find the changefeed owner capture
```
Comment (CharlesCheung96) [inline cdc/api/middleware/middleware.go:112]: Whether it is possible to directly forward to the node where the owner resides?

File: cdc/api/middleware/middleware.go [hunk 3]
```diff
@@ -131,17 +124,48 @@ func ForwardToChangefeedOwnerMiddleware(p capture.Capture,
...
+		changefeedCaptureOwner := controller.GetChangefeedOwnerCaptureInfo(changefeedID)
+		if changefeedCaptureOwner.ID == info.ID {
+			return
+		}
 		api.ForwardToCapture(ctx, info.ID, changefeedCaptureOwner.AdvertiseAddr)
 		ctx.Abort()
 	}
 }
 
+func handleRequestIfIsChnagefeedOwner(ctx *gin.Context, p capture.Capture, changefeedID model.ChangeFeedID) bool {
```
Comment (3AceShowHand) [inline cdc/api/middleware/middleware.go:152]: typo in the function name

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/api/middleware/middleware.go (hunk 2 above)
Comment (sdojjy) [reply cdc/api/middleware/middleware.go:112]: no,  I think only the controller know where is the changefeed owner

**Review by asddongmen — APPROVED** [review]

**Pull request merged**